app/src/main/java/com/android/sample/feature/camera/RequestCameraPermission.kt

```diff
@@ -0,0 +1,47 @@
+package com.android.sample.feature.camera
+
+import androidx.compose.foundation.layout.Box
+import androidx.compose.foundation.layout.fillMaxHeight
+import androidx.compose.foundation.layout.fillMaxSize
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.material3.Button
+import androidx.compose.material3.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import com.google.accompanist.permissions.ExperimentalPermissionsApi
+import com.google.accompanist.permissions.PermissionState
+import com.google.accompanist.permissions.PermissionStatus
+
+/**
+ * @param onPermissionGranted
+ * @return Unit
+ *
+ * Function to request camera permission
+ *
+ * @see RequestCameraPermission
+ * @author based on https://google.github.io/accompanist/permissions/
+ */
+@OptIn(ExperimentalPermissionsApi::class)
+@Composable
+fun RequestCameraPermission(
+    permissionState: PermissionState,
```

Comment:
After some [investigation](https://google.github.io/accompanist/api/permissions/com.google.accompanist.permissions/-permission-state/index.html), I found that `val cameraPermissionState = rememberPermissionState(Manifest.permission.CAMERA)` shouldn’t be null at any point. Instead, it consistently returns a `PermissionState` object, which can have a status of either `PermissionStatus.Denied` or `PermissionStatus.Granted`. We should ensure appropriate handling for each status. 

However, if there is a source indicating otherwise, could you kindly point me to it?

Reply:
Hey, yeah I wrongly explained what I wanted to say, but basically there is apparently some bug that arises sometimes on android when you flip the phone, some states are wrongly initialized and it can make the app crash. For the moment I would say it is not that big for the moment because we blocked the switch to wide screen but maybe if we enable it later we would need to implement this. 

Here is the source of this info:
https://github.com/jstubing/PermissionsTest